packages/m2m-gateway/test/api/agreements/createAgreement.test.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { describe, it, expect, vi } from "vitest";
+import { generateId } from "pagopa-interop-models";
+import { generateToken } from "pagopa-interop-commons-test";
+import { AuthRole, authRole } from "pagopa-interop-commons";
+import request from "supertest";
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
+import { api, mockAgreementService } from "../../vitest.api.setup.js";
+import { appBasePath } from "../../../src/config/appBasePath.js";
+import { getMockedApiAgreement } from "../../mockUtils.js";
+import { toM2MGatewayApiAgreement } from "../../../src/api/agreementApiConverter.js";
+import {
+  missingMetadata,
+  resourcePollingTimeout,
+} from "../../../src/model/errors.js";
+
+describe("POST /agreements router test", () => {
+  const mockAgreementSeed: m2mGatewayApi.AgreementSeed = {
+    eserviceId: generateId(),
+    descriptorId: generateId(),
+  };
+
+  const mockApiAgreement = getMockedApiAgreement();
+  const mockM2MAgreementResponse: m2mGatewayApi.Agreement =
+    toM2MGatewayApiAgreement(mockApiAgreement.data);
+
+  const makeRequest = async (
+    token: string,
+    body: m2mGatewayApi.AgreementSeed
+  ) =>
+    request(api)
+      .post(`${appBasePath}/agreements`)
+      .set("Authorization", `Bearer ${token}`)
+      .send(body);
+
+  const authorizedRoles: AuthRole[] = [authRole.M2M_ADMIN_ROLE];
+  it.each(authorizedRoles)(
+    "Should return 200 and perform service calls for user with role %s",
+    async (role) => {
+      mockAgreementService.createAgreement = vi
+        .fn()
+        .mockResolvedValue(mockM2MAgreementResponse);
+
+      const token = generateToken(role);
+      const res = await makeRequest(token, mockAgreementSeed);
+
+      expect(res.status).toBe(201);
+      expect(res.body).toEqual(mockM2MAgreementResponse);
+    }
+  );
+
+  it.each(
+    Object.values(authRole).filter((role) => !authorizedRoles.includes(role))
+  )("Should return 403 for user with role %s", async (role) => {
+    const token = generateToken(role);
+    const res = await makeRequest(token, mockAgreementSeed);
+    expect(res.status).toBe(403);
+  });
+
+  it("Should return 400 if passed an invalid agreement seed", async () => {
```

Comment:
Here you could use it.each to test multiple invalid seeds as done for example in `createConsumerDelegation.test.ts` here in M2M gw